### DISCUSSION TITLE
How to intall

### DISCUSSION THREAD

**Discussion by PrtmPhlp** [charmbracelet/bubbletea#529] [General]
Hi, first of all, this looks great. I've been doing shell scripting for some time now and wanted to use something new. After a long error journey I successfully go. My Question is: What now? I really searched for some time for Tutorials on how to install, but I'm completely new to go. Could someone please emplain, how I can install it and maybe how to run the tutorial?
Thanks in advance!

**Comment by knz** [comment]
The tutorial and instalation for `gum` is here: https://github.com/charmbracelet/gum#installation

**Comment by bashbunni** [comment]
Hi @PrtmPhlp, first of all, welcome! 

I'd highly recommend looking at our [YouTube channel](https://youtube.com/c/charmcli) to get started. We're updating it regularly and have more tutorials coming out in the next couple of weeks.

Bubble Tea is a Go library, meaning it will be imported as a package in your go programs. From there, you will be able to leverage its public methods and interfaces to build your TUIs. 

To run our examples, you can clone the repository, then `cd` into the `examples` directory and into whichever example you would like to run. You can either use `go run .` or `go build` then `./example-name` where example-name is the same name as the binary created from the build command.

**Comment by bashbunni** [comment]
As for Go resources for getting started, the [official docs](https://go.dev/doc/code) are great. I'd also recommend [Learn Go With Tests](https://quii.gitbook.io/learn-go-with-tests/)
